internal/controller/nonadminbackupstoragelocation_controller.go

```diff
@@ -0,0 +1,682 @@
+// after confirming the VeleroBackupStorageLocation is deleted
+func (r *NonAdminBackupStorageLocationReconciler) removeNaBSLFinalizerUponVeleroBSLDeletion(ctx context.Context, logger logr.Logger, nabsl *nacv1alpha1.NonAdminBackupStorageLocation) (bool, error) {
+	if !controllerutil.ContainsFinalizer(nabsl, constant.NabslFinalizerName) {
+		logger.V(1).Info("NonAdminBackupStorageLocation finalizer not found")
+		return false, nil
+	}
+
+	controllerutil.RemoveFinalizer(nabsl, constant.NabslFinalizerName)
+	if err := r.Update(ctx, nabsl); err != nil {
+		logger.Error(err, "Failed to remove finalizer")
+		return false, err
+	}
+
+	logger.V(1).Info("NonAdminBackupStorageLocation finalizer removed")
+
+	return false, nil
+}
+
+// initNaBSLCreate initializes creation of the NonAdminBackupStorageLocation object
+func (r *NonAdminBackupStorageLocationReconciler) initNaBSLCreate(ctx context.Context, logger logr.Logger, nabsl *nacv1alpha1.NonAdminBackupStorageLocation) (bool, error) {
+	if nabsl.Status.Phase != constant.EmptyString {
+		logger.V(1).Info("NonAdminBackupStorageLocation Phase already initialized", constant.CurrentPhaseString, nabsl.Status.Phase)
+		return false, nil
+	}
+
+	// Set phase to New
+	if updated := updateNaBSLPhase(&nabsl.Status.Phase, nacv1alpha1.NaBSLPhaseNew); updated {
+		if err := r.Status().Update(ctx, nabsl); err != nil {
+			logger.Error(err, statusBslUpdateError)
+			return false, err
+		}
+		logger.V(1).Info("NonAdminBackupStorageLocation Phase set to New")
+	} else {
+		logger.V(1).Info("NonAdminBackupStorageLocation Phase update skipped", constant.CurrentPhaseString, nabsl.Status.Phase)
+	}
+	return false, nil
+}
+
+// validateNaBSLSpec validates the NonAdminBackupStorageLocation spec
+func (r *NonAdminBackupStorageLocationReconciler) validateNaBSLSpec(ctx context.Context, logger logr.Logger, nabsl *nacv1alpha1.NonAdminBackupStorageLocation) (bool, error) {
+	// Skip validation if not in New phase
+	if nabsl.Status.Phase != nacv1alpha1.NaBSLPhaseNew {
+		logger.V(1).Info("Skipping validation, not in New phase", constant.CurrentPhaseString, nabsl.Status.Phase)
+		return false, nil
+	}
+
+	err := function.ValidateBslSpec(ctx, r.Client, nabsl)
+	if err != nil {
+		updatedPhase := updateNonAdminBslPhase(&nabsl.Status.Phase, nacv1alpha1.NaBSLPhaseUnavailable)
+		updatedCondition := meta.SetStatusCondition(&nabsl.Status.Conditions,
+			metav1.Condition{
+				Type:    string(nacv1alpha1.NonAdminConditionAccepted),
+				Status:  metav1.ConditionFalse,
+				Reason:  "BslSpecValidation",
+				Message: err.Error(),
+			},
+		)
+		if updatedPhase || updatedCondition {
+			if updateErr := r.Status().Update(ctx, nabsl); updateErr != nil {
+				logger.Error(updateErr, statusBslUpdateError)
+				return false, updateErr
+			}
+		}
+		return false, reconcile.TerminalError(err)
+	}
+
+	// Validation successful, update phase and condition
+	updatedPhase := updateNaBSLPhase(&nabsl.Status.Phase, nacv1alpha1.NaBSLPhaseNew)
```

Comment:
this was already done in `initNaBSLCreate`

Reply:
@mateusoliveira43 Fixed